### ISSUE TITLE
scale doesn't broadcast correctly for 3D+t data

### ISSUE THREAD

**Issue by jni** [napari/napari#2967]
## 🐛 Bug

A while back we decided that scale and translate would "broadcast" to bigger arrays when fewer values were given than the data dimension. The use case is a 3D+t dataset, in which the spatial scale is known/important, while the temporal scale is unused. In such cases, it's convenient to write:

```python
viewer.add_image(time_series_4d, scale=(2, 1, 1))
```

This used to work and I thought it was tested, but apparently it's now broken:

```python
import numpy as np
import napari
labels = np.zeros((3, 5, 11, 11), dtype=int)
labels_layer = napari.layers.Labels(
    labels, scale=(2, 1, 1), translate=(5, 5, 5)
)
```

raises

<details>

<summary>

```pytb
ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 4 is different from 5)
```

</summary>

```pytb
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-ba9bf4f8265f> in <module>
----> 1 labels_layer = napari.layers.Labels(
      2     labels, scale=(2, 1, 1), translate=(5, 5, 5)
      3 )

~/projects/napari/napari/layers/labels/labels.py in __init__(self, data, num_colors, properties, color, seed, name, metadata, scale, translate, rotate, shear, affine, opacity, blending, visible, multiscale)
    204         )
    205
--> 206         super().__init__(
    207             data,
    208             rgb=False,

~/projects/napari/napari/layers/image/image.py in __init__(self, data, rgb, colormap, contrast_limits, gamma, interpolation, rendering, iso_threshold, attenuation, name, metadata, scale, translate, rotate, shear, affine, opacity, blending, visible, multiscale)
    288
    289         # Trigger generation of view slice and thumbnail
--> 290         self._update_dims()
    291
    292     def _new_empty_slice(self):

~/projects/napari/napari/layers/base/base.py in _update_dims(self, event)
    542         self._ndim = ndim
    543
--> 544         self.refresh()
    545
    546     @property

~/projects/napari/napari/layers/base/base.py in refresh(self, event)
    944         """Refresh all layer data based on current view slice."""
    945         if self.visible:
--> 946             self.set_view_slice()
    947             self.events.set_data()
    948             self._update_thumbnail()

~/projects/napari/napari/layers/base/base.py in set_view_slice(self)
    808     def set_view_slice(self):
    809         with self.dask_optimized_slicing():
--> 810             self._set_view_slice()
    811
    812     @abstractmethod

~/projects/napari/napari/layers/image/image.py in _set_view_slice(self)
    517
    518         # Check if requested slice outside of data range
--> 519         indices = np.array(self._slice_indices)
    520         extent = self._extent_data
    521         if np.any(

~/projects/napari/napari/layers/base/base.py in _slice_indices(self)
    610     def _slice_indices(self):
    611         """(D, ) array: Slice indices in data coordinates."""
--> 612         inv_transform = self._data_to_world.inverse
    613
    614         if self.ndim > self._ndisplay:

~/projects/napari/napari/layers/base/base.py in _data_to_world(self)
    994         affine * (rotate * shear * scale + translate)
    995         """
--> 996         return self._transforms[1:3].simplified
    997
    998     def _update_draw(self, scale_factor, corner_pixels, shape_threshold):

~/projects/napari/napari/utils/transforms/transforms.py in simplified(self)
    123             return self[0]
    124         else:
--> 125             return tz.pipe(self[0], *[tf.compose for tf in self[1:]])
    126
    127     def set_slice(self, axes: Sequence[int]) -> 'TransformChain':

~/conda/envs/all/lib/python3.8/site-packages/toolz/functoolz.py in pipe(data, *funcs)
    632     """
    633     for func in funcs:
--> 634         data = func(data)
    635     return data
    636

~/projects/napari/napari/utils/transforms/transforms.py in compose(self, transform)
    458     def compose(self, transform: 'Affine') -> 'Affine':
    459         """Return the composite of this transform and the provided one."""
--> 460         affine_matrix = self.affine_matrix @ transform.affine_matrix
    461         return Affine(affine_matrix=affine_matrix)
    462

ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 4 is different from 5)
```

</details>

This works, by the way:

```python
import numpy as np
import napari
labels = np.zeros((3, 5, 11, 11), dtype=int)
labels_layer = napari.layers.Labels(
    labels, scale=(1, 2, 1, 1), translate=(5, 5, 5)
)
```

Notice that "translate" is still 3 elements long, so the behavior there is as I expected.

**Comment by sofroniewn** [comment]
cc @andy-sweet

**Comment by andy-sweet** [comment]
This should be fixed by [PR #2990](https://github.com/napari/napari/pull/2990), so I'm going to pick this one up too. I assume that the first dimension in the example above represents time, so padding the leading dimensions of scale and translate with 1 and 0 respectively acts as desired unlike the related #2973

**Comment by andy-sweet** [comment]
Fixed with PR #2990
